projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.js

```diff
@@ -109,6 +109,34 @@
 				}
 			);
 		}
+
+		document.querySelectorAll('li.wp-has-submenu.wp-not-current-submenu').forEach( function (el) {
+			const submenu = el.querySelector('.wp-submenu');
+
+			el.addEventListener('mouseover', function() {
+				submenu.style.display = null;
+				submenu.style.top = '-1px';
+				if ( ! isElementInViewport( submenu ) ) {
+					submenu.style.top = ( 34 - submenu.clientHeight ) + 'px';
+				}
+			} );
+
+			el.addEventListener('mouseleave', function() {
+				submenu.style.display = 'none';
+				submenu.style.top = null;
+			} );
+		} );
+	}
+
+	function isElementInViewport (el) {
```

Comment:
Nit: the whitespace here also feels incorrect.
```suggestion
	function isElementInViewport( el ) {
```